doc/source/data/working-with-images.rst

```diff
@@ -0,0 +1,267 @@
+
+            print(ds.schema())
+
+        .. testoutput::
+
+            Column  Type
+            ------  ----
+            image   numpy.ndarray(shape=(32, 32, 3), dtype=uint8)
+            label   int64
+
+
+For more information on creating datasets, see :ref:`Loading Data <loading_data>`.
+
+.. _transforming_images:
+
+Transforming images
+-------------------
+
+To transform images, call :meth:`~ray.data.Dataset.map` or
+:meth:`~ray.data.Dataset.map_batches`.
+
+.. testcode::
+
+    from typing import Any, Dict
+    import numpy as np
+    import ray
+
+    def increase_brightness(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+        batch["image"] = np.clip(batch["image"] + 4, 0, 255)
+        return batch
+
+    ds = (
+        ray.data.read_images("s3://anonymous@ray-example-data/batoidea/JPEGImages")
+        .map_batches(increase_brightness)
+    )
+
+For more information on transforming data, see
+:ref:`Transforming data <transforming_data>`.
+
+.. _performing_inference_on_images:
+
+Performing inference on images
+------------------------------
+
+To perform inference with a pre-trained model, first load and transform your data.
+
+.. testcode::
+
+    from typing import Any, Dict
+    from torchvision import transforms
+    import ray
+
+    def transform_image(row: Dict[str, Any]) -> Dict[str, Any]:
+        transform = transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Resize((32, 32))
+        ])
+        row["image"] = transform(row["image"])
+        return row
+
+    ds = (
+        ray.data.read_images("s3://anonymous@ray-example-data/batoidea/JPEGImages")
+        .map(transform_image)
+    )
+
+Next, implement a callable class that sets up and invokes your model.
+
+.. testcode::
+
+    import torch
+    from torchvision import models
+
+    class ImageClassifier:
+        def __init__(self):
+            weights = models.ResNet18_Weights.DEFAULT
+            self.model = models.resnet18(weights=weights)
+            self.model.eval()
+
+        def __call__(self, batch):
+            inputs = torch.from_numpy(batch["image"])
+            with torch.inference_mode():
+                outputs = self.model(inputs)
+            return {"class": outputs.argmax(dim=1)}
+
+Finally, call :meth:`Dataset.map_batches() <ray.data.Dataset.map_batches>`.
+
+.. testcode::
+
+    predictions = ds.map_batches(
+        ImageClassifier,
+        compute=ray.data.ActorPoolStrategy(size=2),
+        batch_size=4
+    )
+    predictions.show(3)
+
+.. testoutput::
+
+    {'class': 118}
+    {'class': 153}
+    {'class': 296}
+
+For more information on performing inference, see
+:ref:`End-to-end: Offline Batch Inference <batch_inference_home>`
+and :ref:`Transforming batches with actors <transforming_data_actors>`.
+
+.. _saving_images:
+
+Saving images
+-------------
+
+Save images in Parquet or Numpy files. Other formats aren't supported.
```

Comment:
Do we need to say other formats aren't supported?

Reply:
Let's just link to the input/output ref instead?

Reply:
What would users find in the API references other than `write_parquet` and `write_numpy`?

Reply:
Discussed offline -- `write_json` also works. Will update accordingly